src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions_SharedWithCodeStyle.cs

```diff
@@ -135,16 +143,17 @@ public static bool IsKind(this SyntaxNode node, SyntaxKind kind1, SyntaxKind kin
                 return token.LeadingTrivia;
             }
 
+            Contract.ThrowIfTrue(sourceText == null && includePreviousTokenTrailingTriviaOnlyIfOnSameLine, "If we are including previous token trailing trivia, we need the text too.");
```

Comment:
ick... do people really call this with null?  woudl love to make it non-null and non-optional.  if necessary, introducing a different helper if we really need the case where null/false is supported.

Reply:
I didn't look terribly hard at this helper but it's used in a lot of strange ways. I walked away on this one...